README.md

```diff
@@ -2,86 +2,103 @@
+* `serializer` (function, default `el => el.innerHTML`) controls how the the `content` property returned by the `parse()` method is produced from the root DOM element. It may be useful to specify the `serializer` as the identity function (`el => el`) to obtain a DOM element instead of a string for `content` if you plan to process it further.
 
-```
+### `parse()`
+
+Returns an object containing the following properties:
+
+* `title`: article title;
+* `content`: HTML string of processed article content;
+* `textContent`: text content of the article, with all the HTML tags removed;
+* `length`: length of an article, in characters;
+* `excerpt`: article description, or short excerpt from the content;
+* `byline`: author metadata;
+* `dir`: content direction;
+* `siteName`: name of the site.
+
+The `parse()` method works by modifying the DOM. This removes some elements in the web page, which may be undesirable. You can avoid this by passing the clone of the `document` object to the `Readability` constructor:
+
+```js
 var documentClone = document.cloneNode(true); 
 var article = new Readability(documentClone).parse();
 ```
 
-## Usage from Node.js
+### `isProbablyReaderable(document, options)`
+
+A quick-and-dirty way of figuring out if it's plausible that the contents of a given document are suitable for processing with Readability. It is likely to produce both false positives and false negatives. The reason it exists is to avoid bogging down a time-sensitive process (like loading and showing the user a webpage) with the complex logic in the core of Readability. Improvements to its logic (while not deteriorating its performance) are very welcome.
+
+The `options` object accepts a number of properties, all optional:
+
+* `minContentLength` (number, default `140`): the minimum node content length used to decide if the document is readerable;
+* `minScore` (number, default `20`): the minumum cumulated 'score' used to determine if the document is readerable;
+* `visibilityChecker` (function, default `isNodeVisible`): the function used to determine if a node is visible;
+
+The function returns a boolean corresponding to whether or not we suspect `Readability.parse()` will suceeed at returning an article object. Here's an example:
+
+```js
+/* 
+    Only instantiate Readability  if we suspect 
+    the `parse()` method will produce a meaningful result.
+*/
+if (isProbablyReaderable(document)) {
+    let article = new Readability(document).parse();
+}
+```
+
+## Node.js usage
 
 Readability is available on npm:
 
 ```bash
 npm install @mozilla/readability
 ```
 
-In Node.js, you won't generally have a DOM document object. To obtain one, you can use external
-libraries like [jsdom](https://github.com/jsdom/jsdom). While this repository contains a parser of
-its own (`JSDOMParser`), that is restricted to reading XML-compatible markup and therefore we do
-not recommend it for general use.
```

Comment:
I've removed the mention of `JSDOMParser` — if I understand correctly, people don't actually need to know that it exists. It may be useful, however, to add a recipe for parsing XHTML with `jsdom`? I don't remember exactly, but by default it might choke on the weirder things in XHTML (`CDATA` sections, etc.)